### ISSUE TITLE
Suggest to use `if` and `when` as expression when applicable

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#6233]
## Expected Behavior of the rule
Prohibit usage of `if` or `when` like below

Non-compliant code
```
fun test(foo: Int): Int {
   if (foo > 0) {
      return 0
   } else {
      return 1
   }
}
```
(above code is for `if` used for returning value, the same can be applicable for variable assignment or when expression)

Compliant code
```
fun test(foo: Int): Int {
   return if (foo > 0) {
      0
   } else {
      1
   }
}
```

## Context
Many times in my PR reviews in my office project I see these types of usages which generally happens with people who are new to Kotlin. Having this rule will not only improve the code as well as educated the usage of `if` and `when` as an expression

**Comment by BraisGabin** [comment]
I like the idea!

probably we should also check for things like:

```kotlin
   val a: Int
   if (foo > 0) {
      a = 0
   } else {
      a = 1
   }
```

Which should be the name of this rule? Intellij call this action "lift return"/"lift assignment". `LiftExpression`?

**Comment by ErdoganSeref** [comment]
> I like the idea!
> 
> probably we should also check for things like:
> 
> ```kotlin
>    val a: Int
>    if (foo > 0) {
>       a = 0
>    } else {
>       a = 1
>    }
> ```
> 
> Which should be the name of this rule? Intellij call this action "lift return"/"lift assignment". `LiftExpression`?

I suggest the name `StatementCouldBeExpression`.
